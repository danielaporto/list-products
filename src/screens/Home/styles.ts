import styled from 'styled-components';

export const Container = styled.div`
  min-height: 100vh;
  padding: 30px;
`;

export const ContainerSkeleton = styled.div`
  padding: 10px;
  max-width: 300px;
  border-radius: 2px;
`;

export const GridPost = styled.div`
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  display: grid;
  gap: 30px;
`;
