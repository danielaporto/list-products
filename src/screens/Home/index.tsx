import React, { useCallback, useEffect, useState } from 'react';
import { CardPost } from '../../components/CardPost';
import { getAllListProducts } from '../../actions/list-all-products';
import { Container, ContainerSkeleton, GridPost } from './styles';
import './styles.ts';
import { Skeleton } from '../../components/Skeleton';

const Home = () => {
  const [posts, setPosts] = useState([]);
  const [loadPosts, setLoadPosts] = useState(true);

  const handlePostList = useCallback(async () => {
    try {
      const responsePostsAndPhotos = await getAllListProducts();
      setPosts(responsePostsAndPhotos);
      setLoadPosts(false);
      return responsePostsAndPhotos;
    } catch (err) {
      return err;
    }
  }, []);

  useEffect(() => {
    handlePostList();
  }, [handlePostList]);

  return (
    <>
      <Container className="Home">
        <GridPost className="grid-container">
          {!loadPosts
            ? posts.map((post) => <CardPost posts={post}></CardPost>)
            : [...Array(20)].map((_, index) => (
                <ContainerSkeleton key={index}>
                  <Skeleton width="300px" height="474px" radius="2px" />
                </ContainerSkeleton>
              ))}
        </GridPost>
      </Container>
    </>
  );
};

export default Home;
