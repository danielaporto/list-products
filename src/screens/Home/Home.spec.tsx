import React from 'react';
import { render } from '@testing-library/react';
import Home from '.';

describe('<Home />', () => {
  it('should render <Home />', () => {
    render(<Home />);
  });

  it('should match snapshot', () => {
    const { container } = render(<Home />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
