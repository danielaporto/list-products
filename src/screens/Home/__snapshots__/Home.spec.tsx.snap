// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match snapshot 1`] = `
<div
  class="sc-eDvSVe tULga Home"
>
  <div
    class="sc-gKPRtg gPvHCf grid-container"
  >
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
    <div
      class="sc-jSUZER SnhgF"
    >
      <div
        class="sc-iBYQkv kqqXBj"
        height="474px"
        radius="2px"
        width="300px"
      />
    </div>
  </div>
</div>
`;
